aptible/resource_app.go

```diff
@@ -763,91 +763,93 @@ func updateServiceSizingPolicy(ctx context.Context, d *schema.ResourceData, meta
-			serviceSizingPolicyMap := policies[0].(map[string]interface{})
+		serviceId, err := getServiceIdForAppByName(ctx, meta, appID, serviceName)
+		if err != nil {
+			return err
+		}
 
-			serviceId, err := getServiceIdForAppByName(ctx, meta, appID, serviceName)
-			if err != nil {
-				return err
-			}
+		policy, err := getServiceSizingPolicyForService(serviceId, ctx, meta)
+		if err != nil {
+			log.Println(err)
+			return err
+		}
 
-			autoscaling := serviceSizingPolicyMap["autoscaling_type"].(string)
-			delete(serviceSizingPolicyMap, "autoscaling_type")
-			if autoscaling == "horizontal" {
-				delete(serviceSizingPolicyMap, "maximum_memory")
-			} else if autoscaling == "vertical" {
-				// First, remove values without defaults that aren't used in VAS
-				delete(serviceSizingPolicyMap, "min_containers")
-				delete(serviceSizingPolicyMap, "max_containers")
-				delete(serviceSizingPolicyMap, "min_cpu_threshold")
-				delete(serviceSizingPolicyMap, "max_cpu_threshold")
-				// Now ensure other values are actually set
-				if serviceSizingPolicyMap["maximum_memory"] == 0 {
-					delete(serviceSizingPolicyMap, "maximum_memory")
-				}
-			}
-			// Get rid of anything marked as `0` since that is what terraform sets things not set by the user
-			// Also, 0 is not a valid value for any of ServiceSizingPolicy attributes
-			for key, value := range serviceSizingPolicyMap {
-				switch v := value.(type) {
-				case int:
-					if v == 0 {
-						delete(serviceSizingPolicyMap, key)
-					}
-				case float64:
-					if v == 0 {
-						delete(serviceSizingPolicyMap, key)
-					}
-				}
-			}
-			if update {
-				payload := aptibleapi.NewUpdateServiceSizingPolicyRequest()
-				jsonData, _ := json.Marshal(serviceSizingPolicyMap)
-				_ = json.Unmarshal(jsonData, &payload)
-				payload.Autoscaling = &autoscaling
-				payload.SetScalingEnabled(true)
-
-				_, err = client.ServiceSizingPoliciesAPI.UpdateServiceSizingPolicy(ctx, serviceId).UpdateServiceSizingPolicyRequest(*payload).Execute()
-			} else {
-				payload := aptibleapi.NewCreateServiceSizingPolicyRequest()
-				jsonData, _ := json.Marshal(serviceSizingPolicyMap)
-				_ = json.Unmarshal(jsonData, &payload)
-				payload.Autoscaling = &autoscaling
-
-				_, err = client.ServiceSizingPoliciesAPI.CreateServiceSizingPolicy(ctx, serviceId).CreateServiceSizingPolicyRequest(*payload).Execute()
-			}
-			if err != nil {
-				return fmt.Errorf("failed to create autoscaling policy for service %s: %w", serviceName, err)
-			}
-		} else if update {
-			serviceId, err := getServiceIdForAppByName(ctx, meta, appID, serviceName)
-			if err != nil {
-				return err
-			}
-			// In order to delete, we must first see if there's one associated on the API, otherwise we get an error
-			policy, err := getServiceSizingPolicyForService(serviceId, ctx, meta)
-			if err != nil {
-				log.Println(err)
-				return err
-			}
+		// no policy in the schema but policy exists in deploy-api? delete
+		if len(schemaPolicies) == 0 {
```

Comment:
Am I reading the comment wrong? It seems like it describes the condition:

```
len(schemaPolicies) == 0 && policy != nil
```

which is different than the condition tested here (which doesn't look at policy at all). Seems like either the comment or the condition should change.